source/opt/instruction.cpp

```diff
@@ -372,6 +372,11 @@ uint32_t Instruction::GetTypeComponent(uint32_t element) const {
   return subtype;
 }
 
+Instruction* Instruction::InsertBefore(std::unique_ptr<Instruction>&& i) {
```

Comment:
I moved this up to match the declaration order in the header.
(That's a subtle rule/guidance in the coding style guide.)